### PR TITLE
Fix image registry jobs

### PR DESCRIPTION
**Pull request by bbguimaraes**
This may not be the best solution to the problem, but this issue is blocking
the image registry team:

https://github.com/kubernetes/test-infra/pull/7151#pullrequestreview-103766727

The problem here is that a parent job of `test_branch_image_registry_extended`,
`origin_release_with_ecosystem.yml`, adds `openshift,image-registry=master` to
`sync`.  But this is the repository that triggers the job, so it causes the
validation added on the PR mentioned above to fail.

This PR works around the issue by overriding `sync` and removing
`openshift,image-registry=master` from it.  Not the ideal solution, but it
unblocks the folks from image registry while we work on a better one.

@kargakis, @stevekuznetsov: can you verify this?

### CODE REVIEW

**Comment by kargakis** [comment]
Generate the diff so we can figure out whether the --repo call-out to image-registry is actually dropped?

**Comment by bbguimaraes** [comment]
Done.

**Comment by kargakis** [comment]
Ship it!

**Comment by bbguimaraes** [comment]
@kargakis: added explanation to the commit message, removed WIP.

**Comment by kargakis** [comment]
> (edit: I didn't know that removed the LGTM, sorry =)

Yeah, if you repush your changes, lgtm drops. Still

/lgtm

**Comment by kargakis** [comment]
Push out to Jenkins, I am merging

File: sjb/config/test_cases/test_branch_image_registry_extended.yml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 ---
 parent: 'common/test_cases/origin_built_installed_release.yml'
+overrides:
+  sync:
+    - "openshift,openshift-ansible=master"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_image_registry_extended.yml]: This is the correct approach, but we also want to list `openshift,origin` in here

**Pull request merged**